src/main/scala/gitbucket/core/util/JGitUtil.scala

```diff
@@ -659,17 +660,55 @@ object JGitUtil {
     }
   }
 
+  def getCommitLog(git: Git, from: String, to: String): List[CommitInfo] = {
+    def resolveString(name: String): ObjectId = {
+      val objectId = git.getRepository.resolve(name)
+      git.getRepository.open(objectId).getType match {
+        case Constants.OBJ_COMMIT => objectId
+        case Constants.OBJ_TAG =>
+          val ref = git.getRepository.getRefDatabase.findRef(name)
+          git.getRepository.getRefDatabase.peel(ref).getPeeledObjectId
+        case _ => ObjectId.zeroId()
+      }
+    }
+
+    getCommitLog(git, resolveString(from), resolveString(to))
+  }
+
   /**
    * Returns the commit list between two revisions.
+   * This method takes strings as parameters, but does not define what happens when the strings are no valid
+   * commit ids. They can be either invalid syntactically, or they can be not within the repo.
```

Comment:
Just saying from and to must be valid commit ids would be enough to clarify expectation.